packages/rocketchat-livechat/server/lib/requests.js

```diff
@@ -0,0 +1,26 @@
+import LivechatFileUpload from '../lib/LivechatFileUpload';
+
+/* globals FileUpload, WebApp */
+
+WebApp.connectHandlers.use('/livechat-file-upload/', function(req, res, next) {
+
+	const match = /^\/([^\/]+)\/(.*)/.exec(req.url);
+
+	if (match[1]) {
```

Comment:
same as before.. please rewrite this to be more plain.. also rewrite in a way that the success flow is not inside an if statement.